plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -219,6 +221,40 @@ public static ReaderPageSource createPageSource(
             Optional<ParquetWriteValidation> parquetWriteValidation,
             int domainCompactionThreshold,
             OptionalLong estimatedFileSize)
+    {
+        return createPageSource(
+                inputFile,
+                start,
+                length,
+                columns,
+                ImmutableList.of(effectivePredicate),
+                useColumnNames,
+                timeZone,
+                stats,
+                options,
+                parquetWriteValidation,
+                domainCompactionThreshold,
+                estimatedFileSize);
+    }
+
+    /**
+     * This method is available for other callers to use directly.
+     * The results in the resulting page source will match any of the
+     * predicates specified in the list `effectivePredicates`.
+     */
+    public static ReaderPageSource createPageSource(
```

Comment:
It feels like this should be in a separate commit with some tests with effectivePredicates.size > 1.